components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+		and opponent1.type == Opponent.team
+		and opponent2.type == Opponent.team
+
+	match.extradata = {
+		showh2h = showh2h,
+		casters = Table.isNotEmpty(casters) and Json.stringify(casters) or nil,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getCasterInformation(name, flag, displayName)
+	if String.isEmpty(flag) then
+		flag = Variables.varDefault(name .. '_flag')
+	end
+	if String.isEmpty(displayName) then
+		displayName = Variables.varDefault(name .. '_dn')
+	end
+	if String.isEmpty(flag) or String.isEmpty(displayName) then
+		local parent = Variables.varDefault(
+			'tournament_parent',
+			mw.title.getCurrentTitle().text
+		)
+		local pageName = mw.ext.TeamLiquidIntegration.resolve_redirect(name)
+		local data = mw.ext.LiquipediaDB.lpdb('broadcasters', {
+			conditions = '[[page::' .. pageName .. ']] AND [[parent::' .. parent .. ']]',
+			query = 'flag, id',
+			limit = 1,
+		})
+		if type(data) == 'table' and data[1] then
+			flag = String.isNotEmpty(flag) and flag or data[1].flag
+			displayName = String.isNotEmpty(displayName) and displayName or data[1].id
+		end
+	end
+	if String.isNotEmpty(flag) then
+		Variables.varDefine(name .. '_flag', flag)
+	end
+	if String.isEmpty(displayName) then
+		displayName = name
+	end
+	if String.isNotEmpty(displayName) then
+		Variables.varDefine(name .. '_dn', displayName)
+	end
+	return {
+		name = name,
+		displayName = displayName,
+		flag = flag,
+	}
+end
+
+function matchFunctions.getOpponents(args)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = args['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, args.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getTeamIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = STATUS_HAS_SCORE
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+
+			--set Walkover from Opponent status
+			args.walkover = args.walkover or STATUS_TO_WALKOVER[opponent.status]
+
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				args = matchFunctions.getPlayers(args, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	--set resulttype to 'default' if walkover is set
+	if args.walkover then
+		args.resulttype = 'default'
+	end
+
+	local autofinished = String.isNotEmpty(args.autofinished) and args.autofinished or true
+	-- see if match should actually be finished if score is set
+	if isScoreSet and Logic.readBool(autofinished) and not Logic.readBool(args.finished) then
```

Comment:
```suggestion
	if isScoreSet and autoFinished and not Logic.readBool(args.finished) then
```
due to the proposed change above

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/03d7def6eeb00f6cd3a03b0c1e03f7fac3d3e70b